compiler/kernel/op.go

```diff
@@ -42,14 +43,24 @@ var ErrJoinParents = errors.New("join requires two upstream parallel query paths
 type Builder struct {
 	pctx     *op.Context
 	source   *data.Source
-	planners map[dag.Source]from.Planner
+	listers  map[dag.Source]*meta.Lister
+	pools    map[dag.Source]*lake.Pool
+	progress *zbuf.Progress
```

Comment:
After the compilation is done, GC could reclaim the compiler and leave behind the *zbuf.Progress, though we maybe need to make sure the compiler pointer goes away during the query.  It seemed distasteful to me to have runtime data structures point into the (ephemeral) Builder struct.

Reply:
Also, this will need to change when we distribute the scanners as the stats will need to flow back over the network to be consolidated and returned to the client.

Reply:
👍